File: TelemView-SPA/src/app/_models/product.ts
import { Media } from './media';
import { Lecturer } from './lecturer';
import { Student } from './Student';
import { Tag } from './Tag';
import { Course } from './Course';

export interface Product {
        id: number;
        timeStamp: Date;
        description: string;
        title: string;
        brief: string;
        yearOfCreation: number;
        heYearOfCreation: string;
        degree: string;
        showOnHomePage: boolean;
        taskTitle: string;
        taskDescription: string;
        isPublish: boolean;
        taskId: number;
        productTypeTitle: string;
        productTypeId: number;
        organizationTitle: string;
        organizationId: number;
        productUrl: string;
        organizationTypeTitle: string;
        mainPhotoUrl: string;
        lecturers: Lecturer[];
        students: Student[];
        courses: Course[];
        tags?: Tag[];
        media?: Media[];
        newMedia?: Media[];
}

File: TelemView.API/Models/Product.cs
using System;
using System.Collections.Generic;

namespace TelemView.API.Models
{
    public class Product
    {
        public int Id { get; set; }
        public DateTime TimeStamp { get; set; }
        public string Title { get; set; }
        public string Brief { get; set; }
        public string Description { get; set; }
        public int YearOfCreation { get; set; }
        public string HeYearOfCreation { get; set; }
        public string ProductUrl { get; set; }
        public string Degree { get; set; }
        public bool IsApproved { get; set; }
        public bool IsPublish { get; set; } 
        public bool ShowOnHomePage { get; set; }
        public virtual Task Task {get; set;}
        public int TaskId {get; set;}
        public virtual ProductType ProductType {get; set;}
        public int ProductTypeId {get; set;}
        public virtual Organization Organization { get; set; }
        public int OrganizationId { get; set; }
        public virtual ICollection<Media> Media { get; set; }
        public virtual ICollection<ProductStudent> ProductStudents { get; set; }
        public virtual ICollection<ProductTag> ProductsTags { get; set; }
        public virtual ICollection<ProductLecturer> ProductsLecturers { get; set; }
        public virtual ICollection<ProductCourse> ProductsCourses { get; set; }
    }
}